build/lib/locFunc.ts

```diff
@@ -44,3 +52,301 @@ export function packageSingleExtensionStream(name : string): NodeJS.ReadWriteStr
+				}
+				const translatedMainFile = updateMainI18nFile(existingTranslationFolder + '\\main', './main', mainPack);
+
+				this.queue(translatedMainFile);
+				for (let extension in extensionsPacks) {
+					const translatedExtFile = i18n.createI18nFile(`extensions/${extension}`, extensionsPacks[extension]);
+					this.queue(translatedExtFile);
+
+					//handle edge case for 'Microsoft.sqlservernotebook'
+					const adsExtensionId = (extension === 'Microsoft.sqlservernotebook') ? extension : 'Microsoft.' + extension;
+					resultingTranslationPaths.push({ id: adsExtensionId, resourceName: `extensions/${extension}.i18n.json` });
+				}
+				this.queue(null);
+			})
+			.catch((reason) => {
+				this.emit('error', reason);
+			});
+	});
+}
+
+const textFields = {
+	"nameText": 'ads',
+	"displayNameText": 'Azure Data Studio',
+	"publisherText": 'Microsoft',
+	"licenseText": 'SEE SOURCE EULA LICENSE IN LICENSE.txt',
+	"updateText": 'cd ../vscode && npm run update-localization-extension '
+}
+
+//list of extensions from vscode that are to be included with ADS.
+const VSCODEExtensions = [
+	"bat",
+	"configuration-editing",
+	"docker",
+	"extension-editing",
+	"git-ui",
+	"git",
+	"github-authentication",
+	"github",
+	"image-preview",
+	"json-language-features",
+	"json",
+	"markdown-basics",
+	"markdown-language-features",
+	"merge-conflict",
+	"microsoft-authentication",
+	"powershell",
+	"python",
+	"r",
+	"search-result",
+	"sql",
+	"theme-abyss",
+	"theme-defaults",
+	"theme-kimbie-dark",
+	"theme-monokai-dimmed",
+	"theme-monokai",
+	"theme-quietlight",
+	"theme-red",
+	"theme-seti",
+	"theme-solarized-dark",
+	"theme-solarized-light",
+	"theme-tomorrow-night-blue",
+	"typescript-basics",
+	"xml",
+	"yaml"
+];
+
+/**
+ * A heavily modified version of update-localization-extension that runs using local xlf resources, no arguments required to pass in.
+ * It converts a renamed vscode langpack to an ADS one or updates the existing one to use current XLF resources.
+ * It runs this process on all langpacks currently in the ADS i18n folder.
+ * (Replace the ADS langpack with a corresponding vscode langpack renamed to "ads" instead of "vscode" in order to update vscode core strings and extensions)
+ *
+ * It removes the resources of vscode that we do not support, and adds in new i18n json files created from the xlf files in the folder.
+ * It also merges in the sql core XLF strings with the langpack's existing core strings into a combined main i18n json file.
+ *
+ * Remember to change the version of the langpacks to continue from the previous version of the ADS langpack.
+ *
+ * IMPORTANT: after running on 1 or more vscode langpacks:
+ * 1. change the dependencies of vscode to "*"
+ * 2. add an azdata dependency that matches the current version. (E.G."^1.0.0")
+ * 3. replace the changelog and readme with the ones from previous langpacks, and update the changelog if needed.
+*/
+export function refreshLangpacks(): Promise<undefined> {
```

Comment:
Return type of `Promise<void>` seems like it'd be enough, returning just undefined is a bit odd since that's essentially just not returning anything.